File: advent-of-code/aoc2019/scala3/aoc2019/src/main/scala/Day02.scala
import engineer.kovalev.aoc.Input
import engineer.kovalev.aoc.intcode.IntcodeComputer

@main def day02: Unit =
  println(calculate(12, 2))
  for noun <- 0 to 99 do
    for verb <- 0 to 99 do
      if calculate(noun, verb) == 19690720 then
        println(100 * noun + verb)
        return

def calculate(noun: Int, verb: Int): Int =
  val intcode = Input.readCommaSeparatedInts("../../input/02.txt").toArray
  intcode(1) = noun
  intcode(2) = verb
  val computer = IntcodeComputer(intcode)
  computer.execute()
  computer.memory(0)

File: advent-of-code/aoc2019/rust/src/bin/day02.rs
use aoc2019::intcode::VM;

fn main() {
    part1();
    part2();
}

fn part1() {
    let mut vm = VM::load_program(2);
    vm.set(1, 12);
    vm.set(2, 2);
    vm.execute();
    let value = vm.get(0);
    println!("{value}");
}

fn part2() {
    for noun in 0..=99 {
        for verb in 0..=99 {
            let mut vm = VM::load_program(2);
            vm.set(1, noun);
            vm.set(2, verb);
            vm.execute();
            let value = vm.get(0);
            if value == 19690720 {
                println!("{}", 100 * noun + verb);
            }
        }
    }
}
